src/Mvc/test/Mvc.IntegrationTests/ActionParametersIntegrationTest.cs

```diff
@@ -403,6 +403,71 @@ public async Task ActionParameter_ModelPropertyTypeWithNoParameterlessConstructo
             });
             var modelState = testContext.ModelState;
 
+            // Act & Assert
+            var exception = await Assert.ThrowsAsync<InvalidOperationException>(() => parameterBinder.BindModelAsync(parameter, testContext));
+            Assert.Equal(
+                string.Format(
+                    "Could not create an instance of type '{0}'. Model bound complex types must not be abstract or " +
+                    "value types and must have a parameterless constructor. Record types must have a single primary constructor. " +
+                    "Alternatively, set the '{1}' property to a non-null value in the '{2}' constructor.",
+                    typeof(ClassWithNoDefaultConstructor).FullName,
+                    nameof(Class1.Property1),
+                    typeof(Class1).FullName),
+                exception.Message);
+        }
+
+        public record ActionParameter_DefaultValueConstructor(string name = "test", int age = 23);
+
+        [Fact]
+        public async Task ActionParameter_UsesDefaultConstructorParameters()
```

Comment:
Could you explain this test name❔ Looks like it only confirms the bound `Name` value is used when binding a parameter of the given type. Doesn't confirm anything about constructor parameters than I see.